components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/core/ServerActionManagementService.java

```diff
@@ -212,7 +216,9 @@ private ActionResponse buildActionResponse(Action action) {
                         .uri(action.getEndpoint().getUri())
                         .authentication(new AuthenticationTypeResponse()
                                 .type(AuthenticationTypeResponse.TypeEnum.valueOf(action.getEndpoint()
-                                        .getAuthentication().getType().toString()))));
+                                        .getAuthentication().getType().toString()))))
+                .rule((action.getActionRule() != null) ? RuleMapper.toORRuleResponse(action.getActionRule()) :
```

Comment:
Refer to above comment regarding null check.

Reply:
Refer https://github.com/wso2/identity-api-server/pull/770/files#r1907032092